### PR TITLE
Create Transformations2D.py

### PR DESCRIPTION
**Pull request by Yukti-09**
2D Transformations are regularly used in Linear Algebra.

I have added the codes for reflection, projection, scaling, and rotation matrices.

### CODE REVIEW

**Review by eric-wieser — CHANGES_REQUESTED** [review]
Thanks for the contribution, but as already discussed in numpy/numpy#17043 I think this is not a good fit for numpy.

Some variant of these functions _might_ be a better fit for [`scipy.spatial.transforms`](https://docs.scipy.org/doc/scipy/reference/spatial.transform.html)

File: numpy/linalg/Transformations2D.py [hunk 1]
```diff
@@ -0,0 +1,41 @@
+#2D matrices used for linear transformation in linear algebra 
+
+import numpy as np
+import math 
+
+class Transformations:
```
Comment (eric-wieser) [inline numpy/linalg/Transformations2D.py:6]: Some general feedback for if you submit this elsewhere: this class has no `__init__`, and none of the methods use their `self` argument &mdash; this is a strong indication that you should not be using a class at all.

Comment (Yukti-09) [reply numpy/linalg/Transformations2D.py:6]: Thank you! I will definitely do the needful.

**Comment by eric-wieser** [comment]
xref  #17043

**Pull request closed without merging**